runtime/inspector/mod.rs

```diff
@@ -681,20 +328,21 @@ impl InspectorFlags {
   }
 }
 
-struct InspectorSessions {
-  new_incoming:
-    Pin<Box<dyn Stream<Item = Box<DenoInspectorSession>> + 'static>>,
-  handshake: Option<Box<DenoInspectorSession>>,
-  established: FuturesUnordered<Box<DenoInspectorSession>>,
+/// A helper structure that helps coordinate sessions during different
+/// parts of their lifecycle.
+struct SessionContainer {
+  new_incoming: Pin<Box<dyn Stream<Item = Box<WebsocketSession>> + 'static>>,
+  handshake: Option<Box<WebsocketSession>>,
+  established: FuturesUnordered<Box<WebsocketSession>>,
 }
 
-impl InspectorSessions {
+impl SessionContainer {
   fn new(
-    inspector_ptr: *mut DenoInspector,
+    v8_inspector_ptr: *mut v8::inspector::V8Inspector,
```

Comment:
This is a bit unfortunate - I would prefer to pass reference to inspector, but since `new_websocket_rx` is a future that's not really possible and I had to keep the pointer here.

Reply:
I see... it is a stream, and each `WebSocketSession` needs a mutable reference to the V8Inspector.

If you wanted to it'd be possible to wrap the `V8InspectorSession` in an `Rc<RefCell<_>>` and make clones of it in the `.map()` call.

Reply:
Yeah I was thinking about putting `v8::inspector::V8Inspector` inside a `Rc<RefCell<_>>`. I've hit a wall when trying to decouple `WebsocketSession` from `DenoInspector` so I'll try to apply that change in a follow up PR.